Source/SwiftLintFramework/Rules/Lint/UnusedControlFlowLabelRule.swift

```diff
@@ -86,47 +85,85 @@ public struct UnusedControlFlowLabelRule: SubstitutionCorrectableASTRule, Config
         ]
     )
 
-    private static let kinds: Set<StatementKind> = [.if, .for, .forEach, .while, .repeatWhile, .switch]
+    public func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor? {
+        Visitor(viewMode: .sourceAccurate)
+    }
+
+    public func makeRewriter(file: SwiftLintFile) -> ViolationsSyntaxRewriter? {
+        file.locationConverter.map { locationConverter in
+            Rewriter(
+                locationConverter: locationConverter,
+                disabledRegions: disabledRegions(file: file)
+            )
+        }
+    }
+}
+
+private extension UnusedControlFlowLabelRule {
+    final class Visitor: SyntaxVisitor, ViolationsSyntaxVisitor {
+        private(set) var violationPositions: [AbsolutePosition] = []
+
+        override func visitPost(_ node: LabeledStmtSyntax) {
+            if let position = node.violationPosition {
+                violationPositions.append(position)
+            }
+        }
+    }
+
+    private final class Rewriter: SyntaxRewriter, ViolationsSyntaxRewriter {
+        private(set) var correctionPositions: [AbsolutePosition] = []
+        let locationConverter: SourceLocationConverter
+        let disabledRegions: [SourceRange]
+
+        init(locationConverter: SourceLocationConverter, disabledRegions: [SourceRange]) {
+            self.locationConverter = locationConverter
+            self.disabledRegions = disabledRegions
+        }
+
+        override func visit(_ node: LabeledStmtSyntax) -> StmtSyntax {
+            guard let violationPosition = node.violationPosition else {
+                return super.visit(node)
+            }
+
+            let isInDisabledRegion = disabledRegions.contains { region in
+                region.contains(node.positionAfterSkippingLeadingTrivia, locationConverter: locationConverter)
+            }
+
+            guard !isInDisabledRegion else {
+                return super.visit(node)
+            }
 
-    public func validate(file: SwiftLintFile, kind: StatementKind,
-                         dictionary: SourceKittenDictionary) -> [StyleViolation] {
-        return self.violationRanges(in: file, kind: kind, dictionary: dictionary).map { range in
-            StyleViolation(ruleDescription: Self.description,
-                           severity: configuration.severity,
-                           location: Location(file: file, characterOffset: range.location))
+            let newNode = node.statement.withLeadingTrivia(node.leadingTrivia ?? .zero)
+            correctionPositions.append(violationPosition)
+            return visit(newNode).as(StmtSyntax.self) ?? newNode
```

Comment:
this feels weird, but I think we need to call `visit` on the node returned here to handle nested `LabeledStmtSyntax`s